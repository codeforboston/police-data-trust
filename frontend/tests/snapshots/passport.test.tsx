import * as React from "react"
import Passport from "../../pages/passport/index"
import { render, setAuthForTest, userEvent } from "../test-utils"
import { act } from "react-dom/test-utils"

beforeAll(() => setAuthForTest())

it("renders Passport correctly", () => {
  const { container } = render(<Passport />)
  expect(container).toMatchSnapshot()
})

describe("behaviors", () => {
  function renderPage() {
    const r = render(<Passport />)
    const { getByRole } = r

    return {
      ...r,
      address: getByRole("textbox", { name: /street address/i }),
      city: getByRole("textbox", { name: /city or town/i }),
      state: getByRole("combobox", { name: /state/i }),
      zip: getByRole("spinbutton", { name: /zip code/i }),
      reason: getByRole("textbox", { name: /why are you applying for this role/i }),
      submit: getByRole("button", { name: /submit/i })
    }
  }

  it("submits the form", async () => {
    const r = renderPage()
    act(() => {
      userEvent.type(r.address, "123 Park Street")
      userEvent.type(r.city, "Boston")
      userEvent.type(r.state, "MA")
      userEvent.type(r.zip, "02155")
      userEvent.type(r.reason, Array(10).fill("In a partner organization ").join(""))

      userEvent.click(r.submit)
    })

    await expect(r.findByText(/thank you for your submission/i)).resolves.toBeInTheDocument()
  })
})
