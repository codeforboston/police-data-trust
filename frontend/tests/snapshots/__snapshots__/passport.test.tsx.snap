// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Passport correctly 1`] = `
<div>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
  <div>
    <section
      class="enrollmentSection"
    >
      <div
        class="header"
      >
        <img
          alt="A blue circular logo design reads 'National Police Data Coalition'"
          class="logo medium"
          src="/NPDCLogo.svg"
        />
        <h1>
          Passport Account Application
        </h1>
      </div>
      <p
        class="passportIntro"
      >
        Hello 
        <em>
          FirstTest LastTest
        </em>
        , thank you for your continued interest in the National Police Data Coalition.
        <br />
        <br />
        We are able to provide access to legally protected data to users with the appropriate permissions. This form will submit your profile for approval.
      </p>
      <form
        class="passportForm"
      >
        <fieldset>
          <div
            class="defaultInputContainer inputContainer large"
          >
            <label
              for="streetAddressInput"
            >
              Street Address:
            </label>
            <input
              aria-invalid="false"
              aria-required="true"
              class="inputField"
              id="streetAddressInput"
              name="streetAddress"
              type="text"
              value=""
            />
          </div>
          <div
            class="defaultInputContainer inputContainer undefined"
          >
            <label
              for="cityOrTownInput"
            >
              City Or Town:
            </label>
            <input
              aria-invalid="false"
              aria-required="true"
              class="inputField"
              id="cityOrTownInput"
              name="cityOrTown"
              type="text"
              value=""
            />
          </div>
          <div
            class="defaultInputContainer false"
          >
            <label
              for="stateSelectInput"
            >
              State:
            </label>
            <input
              aria-required="true"
              class="stateSelect false"
              id="stateSelectInput"
              list="stateSelectList"
              name="state"
              type="text"
            />
            <datalist
              id="stateSelectList"
            >
              <option
                value="AK"
              >
                Alaska
              </option>
              <option
                value="AL"
              >
                Alabama
              </option>
              <option
                value="AR"
              >
                Arkansas
              </option>
              <option
                value="AS"
              >
                American Samoa
              </option>
              <option
                value="AZ"
              >
                Arizona
              </option>
              <option
                value="CA"
              >
                California
              </option>
              <option
                value="CO"
              >
                Colorado
              </option>
              <option
                value="CT"
              >
                Connecticut
              </option>
              <option
                value="DC"
              >
                District of Columbia
              </option>
              <option
                value="DE"
              >
                Delaware
              </option>
              <option
                value="FL"
              >
                Florida
              </option>
              <option
                value="GA"
              >
                Georgia
              </option>
              <option
                value="GU"
              >
                Guam
              </option>
              <option
                value="HI"
              >
                Hawaii
              </option>
              <option
                value="IA"
              >
                Iowa
              </option>
              <option
                value="ID"
              >
                Idaho
              </option>
              <option
                value="IL"
              >
                Illinois
              </option>
              <option
                value="IN"
              >
                Indiana
              </option>
              <option
                value="KS"
              >
                Kansas
              </option>
              <option
                value="KY"
              >
                Kentucky
              </option>
              <option
                value="LA"
              >
                Louisiana
              </option>
              <option
                value="MA"
              >
                Massachusetts
              </option>
              <option
                value="MD"
              >
                Maryland
              </option>
              <option
                value="ME"
              >
                Maine
              </option>
              <option
                value="MI"
              >
                Michigan
              </option>
              <option
                value="MN"
              >
                Minnesota
              </option>
              <option
                value="MO"
              >
                Missouri
              </option>
              <option
                value="MS"
              >
                Mississippi
              </option>
              <option
                value="MT"
              >
                Montana
              </option>
              <option
                value="NC"
              >
                North Carolina
              </option>
              <option
                value="NE"
              >
                Nebraska
              </option>
              <option
                value="NH"
              >
                New Hampshire
              </option>
              <option
                value="NJ"
              >
                New Jersey
              </option>
              <option
                value="NM"
              >
                New Mexico
              </option>
              <option
                value="NV"
              >
                Nevada
              </option>
              <option
                value="NY"
              >
                New York
              </option>
              <option
                value="ND"
              >
                North Dakota
              </option>
              <option
                value="OH"
              >
                Ohio
              </option>
              <option
                value="OK"
              >
                Oklahoma
              </option>
              <option
                value="OR"
              >
                Oregon
              </option>
              <option
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                value="PR"
              >
                Puerto Rico
              </option>
              <option
                value="RI"
              >
                Rhode Island
              </option>
              <option
                value="SC"
              >
                South Carolina
              </option>
              <option
                value="SD"
              >
                South Dakota
              </option>
              <option
                value="TN"
              >
                Tennessee
              </option>
              <option
                value="TX"
              >
                Texas
              </option>
              <option
                value="UT"
              >
                Utah
              </option>
              <option
                value="VA"
              >
                Virginia
              </option>
              <option
                value="VT"
              >
                Vermont
              </option>
              <option
                value="WA"
              >
                Washington
              </option>
              <option
                value="WI"
              >
                Wisconsin
              </option>
              <option
                value="WV"
              >
                West Virginia
              </option>
              <option
                value="WY"
              >
                Wyoming
              </option>
            </datalist>
          </div>
          <div
            class="defaultInputContainer inputContainer small"
          >
            <label
              for="zipCodeInput"
            >
              Zip Code:
            </label>
            <input
              aria-invalid="false"
              aria-required="true"
              class="inputField"
              id="zipCodeInput"
              name="zipCode"
              type="number"
              value=""
            />
          </div>
        </fieldset>
        <div
          class="defaultInputContainer false"
        >
          <label
            for="responseTextArea"
          >
            Why are you applying for this role?
          </label>
          <textarea
            aria-describedby="responseCounter null"
            aria-required="true"
            class="responseArea"
            cols="52"
            id="responseTextArea"
            name="reponseText"
            rows="7"
          />
          <div
            class="responseSubtext"
          >
            <p
              id="responseCounter"
            >
              0
              /
              500
            </p>
          </div>
        </div>
        <button
          class="primaryButton"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div
        class="callToActionContainer"
      >
        <p>
          Is the publically available data sufficient for your needs?
        </p>
        <a
          class="callToActionLink"
          href="/search"
        >
          Return to dashboard
        </a>
      </div>
    </section>
  </div>
</div>
`;
